pilot/pkg/model/context.go

```diff
@@ -176,36 +179,35 @@ type Proxy struct {
 	// If nil, the default networking/core v2 generator is used. This field can be set
 	// at connect time, based on node metadata, to trigger generation of a different style
 	// of configuration.
-	Generator Generator
+	Generator ResourceGenerator
 
 	// Active contains the list of watched resources for the proxy, keyed by the DiscoveryRequest type.
 	// It is nil if the Proxy uses the default generator
 	Active map[string]*WatchedResource
 }
 
+// VersionNonce holds information about ack/nack status in the protocol.
```

Comment:
Why ? Completely different things, they are not in a struct in the DiscoveryRequest either